config/cmake/modules/FindHYPRE.cmake

```
@@ -38,51 +38,91 @@ if (HYPRE_FOUND OR TARGET HYPRE)
 endif()
 
 if (HYPRE_FETCH OR FETCH_TPLS)
+  # Collect all HYPRE_ENABLE variables and pass them to hypre, assuming they are BOOL.
+  set(HYPRE_CMAKE_OPTIONS "")
+  get_cmake_property(all_vars VARIABLES)
+  foreach(var ${all_vars})
+    if(var MATCHES "^HYPRE_ENABLE")
+      list(APPEND HYPRE_CMAKE_OPTIONS "-D${var}:BOOL=${${var}}")
+    endif()
+  endforeach()
+
   set(HYPRE_FETCH_VERSION 2.33.0)
```

```suggestion
set(HYPRE_FETCH_VERSION 3.0.0)
```

We might want to bump the fetched version here. Please, feel free to commit the change if it makes sense

---

Glad to see growing interest in the fetch functionality!  I'll test this out early next week.

---

I'm doing something similar in hypre inspired by your work, Chris! https://github.com/hypre-space/hypre/pull/1380